rclpy/src/rclpy/_rclpy.c

```diff
@@ -1333,6 +1333,96 @@ rclpy_create_publisher(PyObject * Py_UNUSED(self), PyObject * args)
   return PyCapsule_New(publisher, "rcl_publisher_t", NULL);
 }
 
+/// Destructor for an event.
+void
+_rclpy_destroy_event(PyObject * pycapsule)
+{
+  rcl_event_t * event = PyCapsule_GetPointer(pycapsule, "rcl_event_t");
+  rcl_ret_t ret_event = rcl_event_fini(event);
```

Comment:
Order sounds wrong there, freeing the C underlying object before the python one?

Reply:
This is called automatically as the destructor of the allocated Capsule when it is destroyed by Python https://github.com/aws-robotics/rclpy/pull/6/files#diff-80dda03110b5be606b823d6b0558b5c2R1416

There isn't a Python object for me to free.

Reply:
Unless I have misunderstood your comment - and you mean `event` should be freed first - but I have to send `event` to `fini` before that, so its members can be freed - if I were to free `event` first I couldn't access its members any more to free them.